src/helpers/single-value-providers.ts

```diff
@@ -50,6 +52,25 @@ export class CustomFieldIntegerValueProvider extends SingleValueProvider<number>
     }
 }
 
+export class DaysWithDataValueProvider extends SingleValueProvider<number> {
+    constructor(public dataType: string, public daysInPast: number) {
+        if(daysInPast < 0) {
+            throw new Error('daysInPast must be a non-negative number');
+        }
+        super('days with data');
+    }
+
+    async getValue(): Promise<number | undefined> {
+        const today = new Date();
+        const startDate = startOfDay(subDays(today, this.daysInPast));
+        const data = await queryDailyData(this.dataType, startDate, today);
+        if(!data || data.length === 0) {
+            return 0;
+        }
+        return new Set(Object.keys(data)).size;
```

Comment:
I think you will want to count the non-zero values here.  The key count will just be the days it computed values for, which will probably be all non-zero values, but I'm not sure we enforce that.  Some daily data types could return `0` for unvalued days.

`return Object.keys(data).reduce((daysWithData, dayKey) => daysWithData + (data[dayKey] > 0 ? 1 : 0), 0);`

Reply:
ah good catch!